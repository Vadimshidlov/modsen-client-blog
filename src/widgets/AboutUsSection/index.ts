export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
