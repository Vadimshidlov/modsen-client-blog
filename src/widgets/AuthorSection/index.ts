export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
