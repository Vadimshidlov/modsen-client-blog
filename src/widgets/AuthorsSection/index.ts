export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
