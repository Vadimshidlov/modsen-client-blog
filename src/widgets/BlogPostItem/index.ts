export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
