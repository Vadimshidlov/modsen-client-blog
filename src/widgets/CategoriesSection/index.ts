export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
