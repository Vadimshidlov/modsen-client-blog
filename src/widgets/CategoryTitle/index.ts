export { CategoryTitle } from '@/widgets/CategoryTitle/ui/CategoryTitle';
