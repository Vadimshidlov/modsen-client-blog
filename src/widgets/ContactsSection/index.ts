export { ContactsSection } from '@/widgets/ContactsSection/ui/ContactsSection';
