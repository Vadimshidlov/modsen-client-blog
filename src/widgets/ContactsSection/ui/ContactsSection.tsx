import React from 'react';
import container from '@/shared/styles/container.module.scss';
import { Inter } from 'next/font/google';
import { ConversationForm } from '@/features/ConversationForm';
import { useTranslations } from 'next-intl';
import styles from './ContactsSection.module.scss';

const inter = Inter({
  subsets: ['latin'],
  weight: '900',
  variable: '--font-inter',
});

const interRegular = Inter({
  subsets: ['latin'],
  weight: '400',
  variable: '--font-inter',
});

export function ContactsSection() {
  const t = useTranslations('ContactsSection');

  return (
    <section className={`${styles.contactsSection} ${container.container}`}>
      <div className={styles.contactsTitleBlockContainer}>
        <h3 className={`${styles.contactsSubTitle} ${inter.variable}`}>{t('subTitle')}</h3>
        <h2 className={styles.contactsTitle}>{t('title')}</h2>
        <span className={styles.contactsText}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
          ut labore et dolore magna aliqua. Ut enim ad minim.
        </span>
      </div>
      <div className={styles.contactsSheduleBlockContainer}>
        <div className={styles.scheduleBlock}>
          <span className={`${styles.scheduleSubTitle} ${interRegular.variable}`}>
            {t('scheduleTitle')}
          </span>
          <h4 className={styles.scheduleTitle}>{t('scheduleDays')}</h4>
          <h4 className={styles.scheduleTitle}>{t('scheduleHours')}</h4>
          <span className={styles.scheduleText}>{t('scheduleSubtitle')}</span>
        </div>
        <div className={styles.scheduleBlock}>
          <span className={`${styles.scheduleSubTitle} ${interRegular.variable}`}>
            {t('scheduleContactsTitle')}
          </span>
          <h4 className={styles.scheduleTitle}>020 7993 2905</h4>
          <span className={styles.scheduleText}>hello@finsweet.com</span>
        </div>
      </div>
      <ConversationForm />
    </section>
  );
}
