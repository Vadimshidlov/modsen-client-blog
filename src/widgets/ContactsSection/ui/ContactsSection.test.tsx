import { render } from '@testing-library/react';
import { ContactsSection } from '@/widgets/ContactsSection';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('ChooseCategorySection tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<ContactsSection />);
    expect(container).toMatchSnapshot();
  });
});
