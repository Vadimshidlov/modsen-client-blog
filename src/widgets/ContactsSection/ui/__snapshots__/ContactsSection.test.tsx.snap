// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseCategorySection tests should matches snapshot 1`] = `
<div>
  <section
    class="contactsSection container"
  >
    <div
      class="contactsTitleBlockContainer"
    >
      <h3
        class="contactsSubTitle variable"
      >
        Mocked subTitle
      </h3>
      <h2
        class="contactsTitle"
      >
        Mocked title
      </h2>
      <span
        class="contactsText"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim.
      </span>
    </div>
    <div
      class="contactsSheduleBlockContainer"
    >
      <div
        class="scheduleBlock"
      >
        <span
          class="scheduleSubTitle variable"
        >
          Mocked scheduleTitle
        </span>
        <h4
          class="scheduleTitle"
        >
          Mocked scheduleDays
        </h4>
        <h4
          class="scheduleTitle"
        >
          Mocked scheduleHours
        </h4>
        <span
          class="scheduleText"
        >
          Mocked scheduleSubtitle
        </span>
      </div>
      <div
        class="scheduleBlock"
      >
        <span
          class="scheduleSubTitle variable"
        >
          Mocked scheduleContactsTitle
        </span>
        <h4
          class="scheduleTitle"
        >
          020 7993 2905
        </h4>
        <span
          class="scheduleText"
        >
          hello@finsweet.com
        </span>
      </div>
    </div>
    <form
      action="#"
      class="conversationForm"
      data-testid="conversation-form"
    >
      <label
        for="userName"
      >
        <input
          class="textInput variable"
          data-testid="userName-input"
          id="userName"
          name="userName"
          placeholder="Mocked namePlaceholder"
          type="text"
          value=""
        />
      </label>
      <label
        for="email"
      >
        <input
          class="textInput variable"
          data-testid="email-input"
          id="email"
          name="email"
          placeholder="Mocked emailPlaceholder"
          type="email"
          value=""
        />
      </label>
      <div
        class="selectCategoryWrapper"
      >
        <select
          class="selectCategory"
          data-testid="conversation-select"
          name="query"
        >
          <option
            class="select__item"
            value="Business"
          >
            Business
          </option>
          <option
            class="select__item"
            value="Startup"
          >
            Startup
          </option>
          <option
            class="select__item"
            value="Economy"
          >
            Economy
          </option>
          <option
            class="select__item"
            value="Technology"
          >
            Technology
          </option>
        </select>
        <img
          alt="select-icon"
          class="selectIcon"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <label
        for="message"
      >
        <textarea
          class="textInput conversationMessage"
          data-testid="message-input"
          id="message"
          name="message"
          placeholder="Mocked messagePlaceholder"
        />
      </label>
      <button
        class="button fullWidth"
        type="submit"
      >
        Mocked buttonText
      </button>
    </form>
  </section>
</div>
`;
